NuKeeper.Update/Process/NuGetFileRestoreCommand.cs

```diff
@@ -14,50 +14,69 @@ public class NuGetFileRestoreCommand : IFileRestoreCommand
     {
         private readonly INuKeeperLogger _logger;
         private readonly INuGetPath _nuGetPath;
+        private readonly IMonoExecutor _monoExecutor;
         private readonly IExternalProcess _externalProcess;
 
         public NuGetFileRestoreCommand(
             INuKeeperLogger logger,
             INuGetPath nuGetPath,
+            IMonoExecutor monoExecutor,
             IExternalProcess externalProcess)
         {
             _logger = logger;
             _nuGetPath = nuGetPath;
+            _monoExecutor = monoExecutor;
             _externalProcess = externalProcess;
         }
 
         public async Task Invoke(FileInfo file, NuGetSources sources)
         {
             _logger.Normal($"Nuget restore on {file.DirectoryName} {file.Name}");
 
-            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                _logger.Normal("Cannot run NuGet.exe file restore as OS Platform is not Windows");
-                return;
-            }
-
             var nuget = _nuGetPath.Executable;
 
             if (string.IsNullOrWhiteSpace(nuget))
             {
-                _logger.Normal("Cannot find NuGet exe for solution restore");
+                _logger.Normal("Cannot find NuGet.exe for solution restore");
                 return;
             }
 
             var sourcesCommandLine = sources.CommandLine("-Source");
 
             var restoreCommand = $"restore {file.Name} {sourcesCommandLine}  -NonInteractive";
 
-            var processOutput = await _externalProcess.Run(file.DirectoryName, nuget,
-                restoreCommand, ensureSuccess: false);
+            ProcessOutput processOutput;
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (await _monoExecutor.CanRun())
+                {
+                    processOutput = await _monoExecutor.Run(file.DirectoryName,
+                        nuget,
+                        restoreCommand,
+                        ensureSuccess: false);
+                }
+                else
+                {
+                    _logger.Normal("Cannot run NuGet.exe. It requires either Windows OS Platform or Mono installation");
```

Comment:
`_logger.Error` would probably fit this scenario better.

Reply:
Just blindly replicated the old behavior, but I think it makes more sense this way.